### DISCUSSION TITLE
API /?limit=0 not really as expected

### DISCUSSION THREAD

**Discussion by benjaminmonnier** [netbox-community/netbox#17589] [Help Wanted!]
Hello, 

when I'm trying to pass /api/ipam/vlans/?limit=0, I receive this response : 
```
{
    "count": 1262,
    "next": "https://netbox-lab.demo/api/ipam/vlans/?limit=1000&offset=1000",
...
}
```

How can I print all my 1262 item on one page ?

Thank you.
I'm using Netbox 3.5.4

**Comment by candlerb** [comment]
The [documentation](https://netboxlabs.com/docs/netbox/en/stable/integrations/rest-api/#pagination) is accurate, although it's easy to misread on first glance.

To permit `limit=0` to return all items, you have to set `MAX_PAGE_SIZE=0` in `configuration.py`.  If you don't do that, then the result is clamped to MAX_PAGE_SIZE (which defaults to 1000).

So you have two choices:
1. Set `MAX_PAGE_SIZE=0` in your configuration.py; or
2. Modify your client code to loop over the API, passing the "next" URL to get the next block - that is, make multiple HTTP calls. Displaying them "on one page" is the job of the application which is calling the API. The end user should not see the API calls made behind the scenes.

Source code reference:

```
class OptionalLimitOffsetPagination(LimitOffsetPagination):
    """
    Override the stock paginator to allow setting limit=0 to disable pagination for a request. This returns all objects
    matching a query, but retains the same format as a paginated request. The limit can only be disabled if
    MAX_PAGE_SIZE has been set to 0 or None.
    """

...

                # Enforce maximum page size, if defined
                MAX_PAGE_SIZE = get_config().MAX_PAGE_SIZE
                if MAX_PAGE_SIZE:
                    return MAX_PAGE_SIZE if limit == 0 else min(limit, MAX_PAGE_SIZE)
                return limit
```
